azure-toolkit-libs/azure-toolkit-containerapps-lib/src/main/java/com/microsoft/azure/toolkit/lib/containerapps/task/DeployContainerAppTask.java

```diff
@@ -78,6 +84,22 @@ private void addCreateAppEnvironmentTaskIfNecessary(@Nonnull final ContainerApps
         }
     }
 
+    //todo(ruoyuwang): check new ACR logic here. Reuse the existing ACR if it exists
+    private void addCreateContainerRegistryTaskIfNecessary(@Nonnull final ContainerRegistryConfig containerRegistryConfig) {
+        final ContainerRegistry registry = Azure.az(AzureContainerRegistry.class).registry(containerRegistryConfig.getSubscriptionId())
```

Comment:
Maybe we can assume the customer will work in a single subscription in most of the cases. So we may have some kind of fallback logic for subscription configuration.

If the customer didn't provide the container registry subscription, we can default to the ACA subscription.

Reply:
The registry default sub and rg will be set to the app sub and rg if not specified.